### PR TITLE
🥔✨ `Space`: `Member` may set `Domain`

### PR DESCRIPTION
**Pull request by KellyAH**
- For https://github.com/zinc-collective/convene/issues/1513

After setting a domain on Space configure page, the Space#update will redirect to that new domain and encounter the issue below. Thus users cannot load the Space configure page to see the domain setting on their space.

e.g. in below example, space.branded_domain = "games.local"

Screenshot 2023-06-25 at 11 45 48 AM

https://github.com/zinc-collective/convene/assets/50284/9945c53e-24cb-4331-a8a2-45ec13569584

### CODE REVIEW

**Comment by zspencer** [comment]
TBH, the cheap-and-easy thing would be to add the `allow_other_hosts: true` on the redirect in the `SpacesController#update` action. It's a little weird, that's for sure; but I think it's probably fine.

File: app/controllers/spaces_controller.rb
Comment (KellyAH) [inline app/controllers/spaces_controller.rb:43]: `space.update` is failing and doing a roll back. But I do not see any error indicating why. 

Screenshot 2023-06-28 at 8 27 52 PM

Comment (anaulin) [reply app/controllers/spaces_controller.rb:43]: You wouldn't see the errors in the server log this way. What you want to do is inspect `space.errors` after the failed update in some way. The typical Rails way is that you have an `else` branch that you use to display the errors when the update is not successful.

Comment (zspencer) [reply app/controllers/spaces_controller.rb:43]: Ideally, the form would also show any validation errors underneath of the field (you can see how to do this by looking at  [`app/views/application/_text_field.html.erb`](https://github.com/zinc-collective/convene/blob/main/app/views/application/_text_field.html.erb), which shows how to use the `_error` partial for this.

**Comment by zspencer** [comment]
@KellyAH - I pushed this over the finish line per your request in Discord. I'll let you decide if you would rather someone other than me from the @zinc-collective/convene-contributors or @zinc-collective/convene-maintainers should do the review!

File: app/controllers/spaces_controller.rb
Comment (zspencer) [inline app/controllers/spaces_controller.rb:45]: Since the `Space#edit` page is so big; it felt weird to be sending them back to the `Space#show` page on change. Also we didn't have any tests about where it was supposed to go which implied to me that it was left unspecified.

File: spec/requests/spaces_controller_request_spec.rb
Comment (zspencer) [inline spec/requests/spaces_controller_request_spec.rb:129]: `true` felt more natural here than `1`

**Review by zspencer — APPROVED** [review]

**Comment by KellyAH** [comment]
Zee's changes LGTM! @anaulin do you think this is okay for me to merge?

**Comment by anaulin** [comment]
@KellyAH yes, go ahead!

In general, if someone already approved it, as far as I'm concerned there is no need to double-check. 🚢

**Pull request merged**
